miio/gateway/gateway.py

```diff
@@ -387,3 +396,43 @@ def get_illumination(self):
             raise GatewayException(
                 "Got an exception while getting gateway illumination"
             ) from ex
+
+    def Register_callback(self, id, callback):
+        """Register a external callback function for updates of this subdevice."""
+        if id in self._registered_callbacks:
+            _LOGGER.error(
+                "A callback with id '%s' was already registed, overwriting previous callback",
+                id,
+            )
+        self._registered_callbacks[id] = callback
+
+    def Remove_callback(self, id):
+        """Remove a external callback using its id."""
+        self._registered_callbacks.pop(id)
+
+    def gateway_push_callback(self, action, params):
+        """Callback from the push server regarding the gateway itself."""
+        for callback in self._registered_callbacks.values():
+            callback(action, params)
+
+    def push_callback(self, source_device, action, params):
+        """Callback from the push server."""
+        if source_device not in self.devices:
+            if source_device == f"lumi.{str(self.device_id)}":
+                self.gateway_push_callback(action, params)
+                return
```

Comment:
I'd move this check before the "parent" condition, make the code cleaner by decreasing nesting.

Reply:
done